Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -281,11 +279,40 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
 
     // MARK: ApplePay
 
-    let paymentAuthorizationData: Signal<PostCampaignPaymentAuthorizationData, Never> = self
+    /*
+
+     Order of operations:
+     1) Apple pay button tapped
+     2) Generate a new payment intent
+     3) Present the payment authorization form
+     4) Payment authorization form calls applePayContextDidCreatePayment with ApplePay params
+     5) Payment authorization form calls paymentAuthorizationDidFinish
+     */
+
+    let newPaymentIntentForApplePay: Signal<String, Never> = self.configureWithDataProperty.signal
+      .skipNil()
+      .takeWhen(self.applePayButtonTappedSignal)
+      .switchMap { initialData in
+        let projectId = initialData.project.graphID
+        let pledgeTotal = initialData.total
+
+        return AppEnvironment.current.apiService
+          .createPaymentIntentInput(input: CreatePaymentIntentInput(
+            projectId: projectId,
+            amountDollars: String(format: "%.2f", pledgeTotal),
+            digitalMarketingAttributed: nil
+          ))
+          .materialize()
+      }
+      .values()
+      .map { $0.clientSecret }
+
+    self.goToApplePayPaymentAuthorization = self
       .configureWithDataProperty
       .signal
       .skipNil()
-      .map { (data: PostCampaignCheckoutData) -> PostCampaignPaymentAuthorizationData? in
+      .combineLatest(with: newPaymentIntentForApplePay)
```

Comment:
This signal shouldn't fire until you have created a payment intent, which only happens after tapping the ApplePay button.